applications/klighd-vscode/src/extension.ts

```diff
@@ -108,51 +109,44 @@ export function activate(context: vscode.ExtensionContext): void {
     context.subscriptions.push(
         vscode.commands.registerCommand(
             command.addActionHandler,
-            (id: string, kind: string, actionHandler: ActionHandlerCallback) => {
-                const extension = extensionMap.get(id);
-                if (!extension) {
-                    vscode.window.showErrorMessage(
-                        `${command.addActionHandler} command called with unknown reference id: ${id}`
-                    );
-                    return;
-                }
+            (kind: string, actionHandler: ActionHandlerCallback) => {
 
                 if (typeof kind !== "string" || typeof actionHandler !== "function") {
                     vscode.window.showErrorMessage(
                         `${command.addActionHandler} command called with invalid arguments. Please refer to the documentation for reference about the correct usage.`
                     );
                     return;
                 }
-
-                extension.addActionHandler(kind, actionHandler);
+                actionHandlers.push({
+                    kind: kind,
+                    actionHandler: actionHandler
+                })
             }
         )
     );
 
     // Command provided for other extensions to dispatch an action if a webview is open
     context.subscriptions.push(
-        vscode.commands.registerCommand(command.dispatchAction, (id: string, action: Action) => {
-            const extension = extensionMap.get(id);
-            if (!extension) {
-                vscode.window.showErrorMessage(
-                    `${command.dispatchAction} command called with unknown reference id: ${id}`
-                );
-                return;
-            }
-
+        vscode.commands.registerCommand(command.dispatchAction, (action: Action) => {
             if (!isAction(action)) {
                 vscode.window.showErrorMessage(
                     `${command.addActionHandler} command called with invalid arguments. Please refer to the documentation for reference about the correct usage.`
                 );
                 return;
             }
-
-            extension.webviews.forEach((webview) => webview.dispatch(action));
+            // TODO: this has not been tested yet if the messenging change works like this.
+            messenger.sendNotification({ method: "diagram/accept" }, { type: 'webview', webviewType: diagramType }, action);
```

Comment:
what exactly is affected by this and what needs to be done to test whether this works?

Reply:
`// Command provided for other extensions to dispatch an action if a webview is open`
As this comment above suggests, this is the feature that I marked as currently not supported in the documentation. To test this we would need to use this feature in another extension, i.e. dispatch some action that should be executed from that other extension. As I don't see a use case currently, I think this is fine to document it being (potentially) not fully supported atm.